utils/httpUtils.ts

```diff
@@ -1,69 +1,129 @@
-    try {
-        const baseURL = process.env.API_URL;
-        const urlPrefix = args.snapshotLinkId ? `/pub/${args.snapshotLinkId}` : "";
-        const response = await fetch(baseURL + urlPrefix + url, {
-            method: 'GET',
-            headers: {
-              'Cookie': context.request.headers.get("cookie"),
-              'Content-Type': 'application/json',
-            },
-          });
-        return response;
-    } catch (e) {
-        return Promise.reject(e.response);
+  try {
+    const baseURL = process.env.API_URL;
+    const nextGenEnabled = await isNextGenEnabled(context);
+    if (nextGenEnabled) {
+      // next gen details
+    } else {
+      const urlPrefix = args.snapshotLinkId
+        ? `/pub/${args.snapshotLinkId}`
+        : "";
+      const response = await fetch(baseURL + urlPrefix + url, {
+        method: "GET",
+        headers: {
+          Cookie: context.request.headers.get("cookie"),
+          "Content-Type": "application/json",
+        },
+      });
+      return response;
     }
+  } catch (e) {
+    return Promise.reject(e.response);
+  }
 };
 
-export const postWithCSRF = async (url: string, body: any, args: any, context: any) => {
-    try {
-        const response = await fetch(process.env.API_URL + url, {
-            method: 'POST',
-            headers: {
-              'Cookie': context.request.headers.get("cookie"),
-              'Content-Type': 'application/json',
-              'X-CSRF-Token': args?.input?.authenticityToken,
-            },
-            body: JSON.stringify(body),
-          });
-        return await response.json();
-    } catch (e) {
-        return Promise.reject(e.response);
+export const postWithCSRF = async (
+  url: string,
+  body: any,
+  args: any,
+  context: any
+) => {
+  try {
+    const nextGenEnabled = await isNextGenEnabled(context);
+    if (nextGenEnabled) {
+      // next gen details
+    } else {
+      const response = await fetch(process.env.API_URL + url, {
+        method: "POST",
+        headers: {
+          Cookie: context.request.headers.get("cookie"),
+          "Content-Type": "application/json",
+          "X-CSRF-Token": args?.input?.authenticityToken,
+        },
+        body: JSON.stringify(body),
+      });
+      return await response.json();
     }
-}
+  } catch (e) {
+    return Promise.reject(e.response);
+  }
+};
 
 export const notFound = (message: string) => {
-    return Promise.reject({
-        status: 404,
-        message: message,
+  return Promise.reject({
+    status: 404,
+    message: message,
+  });
+};
+
+export const getFeatureFlags = async (context: any) => {
+  try {
+    const response = await fetch(process.env.API_URL + "/users/feature_flags", {
+      method: "GET",
+      headers: {
+        Cookie: context.request.headers.get("cookie"),
+        "Content-Type": "application/json",
+      },
     });
-}
+    return await response.json();
+  } catch (e) {
+    return Promise.reject(e.response);
+  }
+};
```

Comment:
This function fetches feature flags from Rails right?  Do we need this?  I thought that we are expecting all feature flags to be passed from the front end.